src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -182,15 +92,36 @@ public void target() {
   }
 
   /** Starts the roller. */
-  private void start() {
+  public void start() {
     mRoller.setSpeed(ROLLER_SPEED);
   }
 
   /** Stops the roller. */
-  private void stop() {
+  public void stop() {
     mRoller.setSpeed(0);
   }
 
+  /** Returns whether roller is at full speed. */
+  public boolean isAtVelocity() {
+    return mRollerVelocity > TARGET_ROLLER_VELOCITY;
+  }
+
+  /** Starts the firing timer and roller.  */
+  public void shootBall() {
+    mFireTimer.start();
+    start();
+  }
+
+  /** Returns whether the firing timer has run longer than the duration. */
+  public boolean isDoneShooting() {
+    if (mFireTimer.get() >= FIRE_DURATION_SECONDS * 1000) {
+      stop();
+      mFireTimer.stop();
+      mFireTimer.reset();
+    }
+    return mFireTimer.get() >= FIRE_DURATION_SECONDS * 1000;
```

Comment:
Why are you comparing twice? Instead use an else, and return false in the else part. Return true in the if.

Reply:
Resolved